controllers/provider-gcp/pkg/controller/worker/machines.go

```diff
@@ -123,6 +123,7 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				"region":             w.worker.Spec.Region,
 				"zone":               zone,
 				"canIpForward":       true,
+				"attachExternalIP":   false,
```

Comment:
As discussed, this will roll all GCP machines in the next reconciliation.
/cc @vlerenc